### PR TITLE
OBDX-21 T3. Create endpoint for order creation messages

### PR DESCRIPTION
**Pull request by chetanmaharishi**
OBDX-21 T3. Create endpoint for order creation messages

### CODE REVIEW

File: grails-app/controllers/org/pih/warehouse/notification/NotificationController.groovy [hunk 1]
```diff
@@ -154,5 +159,200 @@ class NotificationController {
...
+            requisition.name = "DEFAULT"
+        }
+        try {
+            if (oJson.has("origin") && oJson.getString("origin")) {
+                JSONObject jsonObject = oJson.getJSONObject("origin")
+                log.info "Origin location Json:${jsonObject}"
+                Location location = Location.findByLocationNumber(jsonObject.getString("id"))
+                if (!location && grailsApplication.config?.openboxes?.integration?.destinationMapping) {
+                    log.info "Location Mapping::${grailsApplication.config?.openboxes?.integration?.destinationMapping}"
+                    JSONObject destinationMapping = toJson(grailsApplication.config?.openboxes?.integration?.destinationMapping)
```
Comment (jmiranda) [inline grails-app/controllers/org/pih/warehouse/notification/NotificationController.groovy:214]: We should at least break this out into a separate method so we don't repeat the code in both places.
```
getLocationByLocationMapping(jsonObject.getString("id")
```

File: grails-app/controllers/org/pih/warehouse/notification/NotificationController.groovy [hunk 2]
```diff
@@ -154,5 +159,200 @@ class NotificationController {
...
+        if (oJson.has("name")) {
+            requisition.name = oJson.getString("name")
+        } else {
+            requisition.name = "DEFAULT"
+        }
+        try {
+            if (oJson.has("origin") && oJson.getString("origin")) {
+                JSONObject jsonObject = oJson.getJSONObject("origin")
+                log.info "Origin location Json:${jsonObject}"
+                Location location = Location.findByLocationNumber(jsonObject.getString("id"))
```
Comment (jmiranda) [inline grails-app/controllers/org/pih/warehouse/notification/NotificationController.groovy:211]: Shouldn't this be locationNumber now?

**Review by jmiranda — CHANGES_REQUESTED** [review]

**Comment by jmiranda** [comment]
@chetanmaharishi if you don't get to these changes i'll work on the PR at some point during break.

**Comment by chetanmaharishi** [comment]
I am working on these

**Review by jmiranda — APPROVED** [review]

**Pull request merged**